code/modules/fishing/fishing_rod.dm

```diff
@@ -0,0 +1,484 @@
+	inhand_x_dimension = 64
+	inhand_y_dimension = 64
+	force = 8
+	w_class = WEIGHT_CLASS_HUGE
+
+	/// How far can you cast this
+	var/cast_range = 5
+	/// Fishing minigame difficulty modifier (additive)
+	var/difficulty_modifier = 0
+	/// Explaination of rod functionality shown in the ui
+	var/ui_description = "A classic fishing rod, with no special qualities."
+
+	var/obj/item/bait
+	var/obj/item/fishing_line/line
+	var/obj/item/fishing_hook/hook
+
+	/// Currently hooked item for item reeling
+	var/obj/item/currently_hooked_item
+
+	/// Fishing line visual for the hooked item
+	var/datum/beam/hooked_item_fishing_line
+
+	/// Are we currently casting
+	var/casting = FALSE
+
+	/// List of fishing line beams
+	var/list/fishing_lines = list()
+
+	var/default_line_color = "gray"
+
+/obj/item/fishing_rod/Initialize(mapload)
+	. = ..()
+	register_context()
+	register_item_context()
+
+/obj/item/fishing_rod/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	if(src == held_item)
+		if(currently_hooked_item)
+			context[SCREENTIP_CONTEXT_LMB] = "Reel in"
+		context[SCREENTIP_CONTEXT_RMB] = "Modify"
+		return CONTEXTUAL_SCREENTIP_SET
+	return NONE
+
+/obj/item/fishing_rod/add_item_context(obj/item/source, list/context, atom/target, mob/living/user)
+	. = ..()
+	if(currently_hooked_item)
+		context[SCREENTIP_CONTEXT_LMB] = "Reel in"
+		return CONTEXTUAL_SCREENTIP_SET
+	return NONE
+
+/obj/item/fishing_rod/Destroy(force)
+	. = ..()
+	//Remove any leftover fishing lines
+	QDEL_LIST(fishing_lines)
+
+
+/// Catch weight modifier for the given fish_type (or FISHING_DUD), additive
+/obj/item/fishing_rod/proc/fish_bonus(fish_type)
+	return 0
+
+/obj/item/fishing_rod/proc/consume_bait()
+	if(bait)
+		QDEL_NULL(bait)
+		update_icon()
+
+/obj/item/fishing_rod/interact(mob/user)
+	if(currently_hooked_item)
+		reel(user)
+
+/obj/item/fishing_rod/proc/reel(mob/user)
+	//Could use sound here for feedback
+	if(do_after(user, 1 SECONDS, currently_hooked_item))
+		// Should probably respect and used force move later
+		step_towards(currently_hooked_item, get_turf(src))
+		if(get_dist(currently_hooked_item,get_turf(src)) < 1)
+			clear_hooked_item()
+
+/obj/item/fishing_rod/attack_self_secondary(mob/user, modifiers)
+	. = ..()
+	ui_interact(user)
+
+/obj/item/fishing_rod/pre_attack(atom/A, mob/living/user, params)
+	. = ..()
+	/// Reel in if able
+	if(currently_hooked_item)
+		reel(user)
+		return TRUE
+	SEND_SIGNAL(A, COMSIG_PRE_FISHING)
+
+/// Generates the fishing line visual from the current user to the target and updates inhands
+/obj/item/fishing_rod/proc/create_fishing_line(atom/movable/target, target_py = null)
+	var/mob/user = loc
+	if(!istype(user))
+		return
+	var/beam_color = line?.line_color || default_line_color
+	var/datum/beam/fishing_line/fishing_line_beam = new(user, target, icon_state = "fishing_line", beam_color = beam_color, override_target_pixel_y = target_py)
+	fishing_line_beam.lefthand = user.get_held_index_of_item(src) % 2 != 0
```

Comment:
this is honestly such a weird way to check if it is the left hand. Why not just do a direct comparison to check if it equals 1?